src/hacl_x25519.mli

```diff
@@ -35,19 +35,6 @@ type error =
 val pp_error : Format.formatter -> error -> unit
 (** Pretty printer for errors *)
 
-val priv_key_of_cstruct : Cstruct.t -> (secret, error) result
-(** Convert a [Cstruct.t] into a private key. Internally, this only checks that
-    its length is [key_length_bytes]. If that is not the case, returns an error
-    message. *)
-
-val priv_key_to_cstruct : secret -> Cstruct.t
-(** Return the [Cstruct.t] corresponding to a private key. It is always
-    [key_length_bytes] bytes long. *)
-
-val public : secret -> Cstruct.t
-(** Compute the public part corresponding to a private key. Internally, this
-    multiplies the curve's base point by the supplied scalar. *)
-
 val key_exchange : secret -> Cstruct.t -> (Cstruct.t, error) result
 (** Perform Diffie-Hellman key exchange between a private part and a public
     part.
```

Comment:
from TLS1.3 RFC:
```
For these curves, implementations SHOULD use the approach specified
   in [RFC7748] to calculate the Diffie-Hellman shared secret.
   Implementations MUST check whether the computed Diffie-Hellman shared
   secret is the all-zero value and abort if so, as described in
   Section 6 of [RFC7748].  If implementors use an alternative
   implementation of these elliptic curves, they SHOULD perform the
   additional checks specified in Section 7 of [RFC7748].
```

now, does this implementation follow the construction of 7748 (Montgomery ladder)? if so, this should be mentioned.
again, the resulting secret is _not_ truncated AFAICT.

and this function actually checks the second input argument for being not-all-zero, doesn't it (should be documented)?

Reply:
another question: any checks necessary for the received data being a point on the curve, as in p256?

Reply:
Thanks for the feedback!
- I just added that the secret is not truncated;
- if the public part is all zeroes, the internal output will be all zeroes as well, so this case is already accounted for - I added an extra test;
- there are no extra checks to be performed if the function returns `Ok _`.